core/domain/html_cleaner.py

```diff
@@ -364,9 +376,24 @@ def convert_to_text_angular(html_data):
     # Ensure that p tag is not wrapped in p tag.
     for p in soup.findAll('p'):
         if p.parent.name == 'p':
-            p.unwrap()
+            contents = p.parent.contents
```

Comment:
I'm a bit confused by this new code block. Could you explain what it is trying to do?

Reply:
Earlier I followed this approach:
Current html: `<p>hello <p>test </p>case <p>in progress </p></p>`
html after migration: `<p>hello test case in progress</p>`

I have now changed this as follows:
html after migration: `<p>hello </p><p>test </p><p>case </p><p>in progress </p>`
This is same output that we get from the TextAngular demo and also according to the data obtained from dump , I think this will maintain the feel and look for text.

Reply:
OK, let's add a test that demonstrates this functionality specifically? This is a largish change, I would expect some test to change as well.

Reply:
Added test cases, PTAL!